packages/components-library/tests/unit/filters/StringFilter.spec.ts

```diff
@@ -0,0 +1,58 @@
+import { mount, createLocalVue } from '@vue/test-utils'
+import StringFilter from '@/components/filters/StringFilter.vue'
+import { BootstrapVue } from 'bootstrap-vue'
+
+const localVue = createLocalVue()
+localVue.use(BootstrapVue)
+
+describe('StringFilter.vue', () => {
+  let wrapper: any
+  beforeEach(() => {
+    wrapper = mount(StringFilter, {
+      localVue,
+      propsData: {
+        name: 'name',
+        label: 'label',
+        value: 'value'
+      }
+    })
+  })
+
+  it('matches the snapshot', () => {
+    expect(wrapper.element).toMatchSnapshot()
+  })
+
+  it('sets value property on inner input', () => {
+    const inputElement = wrapper.find('input').element as HTMLInputElement
+    expect(inputElement.value).toBe('value')
+  })
+
+  it('emits undefined when clear button is clicked', () => {
+    wrapper.find('button').trigger('click')
+    // @ts-ignore
```

Comment:
Zijn die ts-ignore statements nog nodig?

Reply:
yes it is , the emitted function is not guaranteed to have a input prop, but for the test i do not care about this warning ( in app code i would )